core/src/avm2/globals/date.rs

```diff
@@ -0,0 +1,1373 @@
+//! `Date` class
+
+use crate::avm2::activation::Activation;
+use crate::avm2::class::Class;
+use crate::avm2::method::{Method, NativeMethodImpl};
+use crate::avm2::names::{Namespace, QName};
+use crate::avm2::object::{date_allocator, DateObject, Object, TObject};
+use crate::avm2::string::AvmString;
+use crate::avm2::value::Value;
+use crate::avm2::Error;
+use chrono::{DateTime, Datelike, Duration, FixedOffset, LocalResult, TimeZone, Timelike, Utc};
+use gc_arena::{GcCell, MutationContext};
+use num_traits::ToPrimitive;
+
+enum YearType {
+    Full,
+    Adjust(Box<dyn Fn(i64) -> i64>),
```

Comment:
Is it necessary for this to take a fully boxed function? As far as I can tell, the only place this is actually in use is adjusting two-digit years (e.g. 99 becomes 1999). Would it make more sense to just have `YearType::Full` and `YearType::TwoDigits`? Or are we anticipating other year adjustment schemes in the future?

Reply:
Yeah I agree that having a YearType::TwoDigits would make more sense. I also think that change should be made to the AVM1 version as well

Reply:
So I ended up actually just removing YearType entirely because there really wasn't much point to it. As a replacement I added a `map_year` method, which does the same thing except the year is adjusted immediately, rather than in the `calculate` method.